compiler/noirc_evaluator/src/ssa/opt/remove_bit_shifts.rs

```diff
@@ -148,7 +148,7 @@ impl Context<'_, '_, '_> {
             || {
                 // If we don't know `rhs`'s value then it could be anything up to the number
                 // of bits in the type, e.g. u32 means shifting by up to 32 bits as otherwise we get overflow.
```

Comment:
This comments suggests there is something to do if we do know the value, but it looks like that's no longer the case?

Reply:
I thought so, but in the end the comment is kind of right but the implementation was wrong. If we don't know what `rhs` is then the number of bits we shift will be at most the value's bit size, so for u32 that's 32 (or, well, it could be up to 2^^32 - 1 but after 32 it's an overflow).